src/pages/PlaylistManagementPage/PlaylistManagementPage.js

```diff
@@ -32,23 +33,45 @@ const PlaylistManagementPage = (props) => {
   }, [])
 
   const playlistClickedHandler = (playlistId) => {
+    const playlists = [...collaborativePlaylists]
+    const playlist = playlists.find((playlist) => playlist.id === playlistId)
+
+    if(playlist.selected) {
+      axios
+        .delete(
+          `${process.env.REACT_APP_API_BASE_URL}/playlist/${playlistId}`,
+          { withCredentials: true },
+        )
+        .then((response) => {
+          playlist.selected = !playlist.selected
+          setCollaborativePlaylists(() => playlists)
+        })
+        .catch((err) => console.error('Something went wrong'))
+    } else {
+      axios
+        .post(
+          `${process.env.REACT_APP_API_BASE_URL}/playlist/${playlistId}`,
+          { withCredentials: true },
+        )
+        .then((response) => {
+          playlist.selected = !playlist.selected
```

Comment:
isn't it better to use 'true' as it was before?

Reply:
I implemented the register/unregister logic. So clicking a playlist that is already registered will send the request to unregister the playlist and will remove the green border to tell the user that the playlist is no longer registered.

Reply:
Sure. Do you think there is a way to implement this without so much code duplication? The only thing that changes is the `post` and `delete` part. Is there a way to change the method with an if? Maybe some way to set the same `.then().catch()` on a promise created inside the if?